File: LeetCode/April Challenge/2nd Day/Day_2.cpp
class Solution {
public:
    bool isHappy(int n) 
    {
        set<int> st;
        while(1)
        {
            n=sqsum(n);
            if(n==1)
                return true;
            if(st.find(n)!=st.end())
                return false;
            st.insert(n);
        }
    }
    int sqsum(int n)
    {
        int s=0;
        while(n)
        {
            s+=(n%10)*(n%10);
            n/=10;
        }
        return s;
    }
};

File: LeetCode/April Challenge/2nd Day/Day_2.py
class Solution:
    def isHappy(self, n: int) -> bool:
        notHappy  = set()
        n = str(n)
        while n != '1':
            notHappy.add(n)
            s = 0
            for d in n:
                s += int(d)*int(d)
            n = str(s)
            if n in notHappy:
                return False
        return True
            
